src/onnx/parse_matmul.cpp

```diff
@@ -89,55 +182,84 @@ struct parse_matmul : op_parser<parse_matmul>
             auto s1_lens        = a1->get_shape().lens();
-            instruction_ref ba0 = a0;
-            instruction_ref ba1 = a1;
-            // try broadcasting if dimensions other than last two do not match
-            if(not std::equal(
-                   s0_lens.rbegin() + 2, s0_lens.rend(), s1_lens.rbegin() + 2, s1_lens.rend()))
+
+            if(not is_quant_dot and args.size() > 2)
+            {
+                MIGRAPHX_THROW("PARSE_MATMUL: Bias Args not supported for MatMul");
+            }
+
+            bool has_ba0        = false;
+            bool has_ba1        = false;
+            instruction_ref ba0 = set_bias_arg(args, 2, a0, has_ba0);
+            instruction_ref ba1 = set_bias_arg(args, 3, a1, has_ba1);
+
+            // Only INT8 or UINT8 type currently supported
+            std::set<migraphx::shape::type_t> supported_types = {migraphx::shape::uint8_type,
+                                                                 migraphx::shape::int8_type};
+            const auto a0_type                                = a0->get_shape().type();
+            const auto a1_type                                = a1->get_shape().type();
+            const auto ba0_type                               = ba0->get_shape().type();
+            const auto ba1_type                               = ba1->get_shape().type();
+
+            if(is_quant_dot and
+               (not contains(supported_types, ba0_type) or
+                not contains(supported_types, ba1_type) or not contains(supported_types, a0_type) or
+                not contains(supported_types, a1_type)))
             {
-                auto l0_it = s0_lens.begin() + s0_lens.size() - 2;
-                std::vector<std::size_t> l0_broadcasted_lens(s0_lens.begin(), l0_it);
-                auto l1_it = s1_lens.begin() + s1_lens.size() - 2;
-                std::vector<std::size_t> l1_broadcasted_lens(s1_lens.begin(), l1_it);
-                auto output_lens =
-                    compute_broadcasted_lens(l0_broadcasted_lens, l1_broadcasted_lens);
-                l0_broadcasted_lens = output_lens;
-                l0_broadcasted_lens.insert(l0_broadcasted_lens.end(), l0_it, s0_lens.end());
-                l1_broadcasted_lens = output_lens;
-                l1_broadcasted_lens.insert(l1_broadcasted_lens.end(), l1_it, s1_lens.end());
-                if(s0_lens != l0_broadcasted_lens)
+                MIGRAPHX_THROW("PARSE_MATMULINTEGER: Unsupported type");
+            }
+
+            if(is_quant_dot)
+            {
+                instruction_ref offset_op;
+                if(a0_type == migraphx::shape::uint8_type or a1_type == migraphx::shape::uint8_type)
                 {
-                    ba0 = info.add_instruction(
-                        make_op("multibroadcast", {{"out_lens", l0_broadcasted_lens}}), a0);
+                    offset_op = info.add_literal(
+                        migraphx::literal{migraphx::shape{migraphx::shape::int16_type}, {-128}});
```

Comment:
Better make them `float16`. `int16` is unsupported type and would be converted into `float` by eliminate_data_type

Reply:
Done